<a href="https://colab.research.google.com/github/cjhaster/NRSurrogateCatalog/blob/gh-pages/_sources/events/GW190620_030421.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open in Colab"/></a>

In [1]:
%load_ext autoreload
%autoreload 2
%matplotlib inline

import pandas as pd

pd.set_option("display.max_rows", None, "display.max_columns", None)

In [2]:
import pip

try:
    __import__("nrsur_catalog")
except ImportError:
    pip.main(["install", "nrsur_catalog", "-q", ])


# GW190620_030421

Below are some plots for GW190620_030421 from the NRSurrogate Catalog.

In [3]:
from nrsur_catalog import NRsurResult

nrsur_result = NRsurResult.load("GW190620_030421", cache_dir=".nrsur_catalog_cache")
# you can specify a `cache_dir`: folder where data will be downloaded

## Summary

In [4]:
nrsur_result.summary()

Posterior 90% CI  \
Parameter                                                         
$m_1\ [M_{\odot}]$                  ${84.54}_{-10.32}^{+11.37}$   
$m_2\ [M_{\odot}]$                  ${52.20}_{-12.34}^{+11.93}$   
$\mathcal{M}\ [M_{\odot}]$            ${57.09}_{-6.14}^{+5.30}$   
$q$                                    ${0.62}_{-0.19}^{+0.23}$   
$a_1$                                  ${0.65}_{-0.30}^{+0.23}$   
$a_2$                                  ${0.56}_{-0.36}^{+0.30}$   
$\theta_1$                             ${0.83}_{-0.40}^{+0.45}$   
$\theta_2$                             ${1.13}_{-0.56}^{+0.73}$   
$\chi_{\mathrm{eff}}$                  ${0.32}_{-0.15}^{+0.14}$   
$\chi_{\mathrm{p}}$                    ${0.47}_{-0.21}^{+0.24}$   
$\alpha$                               ${4.27}_{-1.26}^{+0.22}$   
$\delta$                               ${0.44}_{-0.68}^{+0.49}$   
$t_c\ [s]$                    ${1245035079.32}_{-0.02}^{+0.00}$   
$d_L\ [Mpc]$                   ${2966.44}_{-956.83}^{+1071.36}$   
$\phi$                                 ${2.77}_{-1.85}^{+2.29}$   
$\phi$                                 ${5.01}_{-1.89}^{+0.69}$   
$\theta$                               ${1.28}_{-0.15}^{+0.18}$   
$\psi$                                 ${1.43}_{-0.79}^{+0.96}$   
$\phi_{\mathrm{JL}}$                   ${3.04}_{-2.37}^{+2.34}$   
$\phi_{12}$                            ${3.18}_{-2.26}^{+2.18}$   
$\theta_{\mathrm{JN}}$                 ${2.16}_{-1.54}^{+0.55}$   
$m_f\ [M_{\odot}]$                   ${129.48}_{-9.28}^{+9.10}$   
$\chi_f$                               ${0.79}_{-0.07}^{+0.05}$   
$v_{\rm kick}\ [{\rm km/s}]$     ${687.60}_{-458.07}^{+804.75}$   

                                                                        Prior  
Parameter                                                                      
$m_1\ [M_{\odot}]$                                                        $-$  
$m_2\ [M_{\odot}]$                                                        $-$  
$\mathcal{M}\ [M_{\odot}]$    $\text{UniformInComponentsChirpMass} [12, 400]$  
$q$                           $\text{UniformInComponentsMassRatio} [0.17, 1]$  
$a_1$                                              $\text{Uniform} [0, 0.99]$  
$a_2$                                              $\text{Uniform} [0, 0.99]$  
$\theta_1$                                            $\text{Sine} [0, 3.14]$  
$\theta_2$                                            $\text{Sine} [0, 3.14]$  
$\chi_{\mathrm{eff}}$                                                     $-$  
$\chi_{\mathrm{p}}$                                                       $-$  
$\alpha$                                                                  $-$  
$\delta$                                                                  $-$  
$t_c\ [s]$                    $\text{Uniform} [1245035079.21, 1245035079.41]$  
$d_L\ [Mpc]$                     $\text{UniformSourceFrame} [100.0, 10000.0]$  
$\phi$                                             $\text{Uniform} [0, 6.28]$  
$\phi$                                             $\text{Uniform} [0, 6.28]$  
$\theta$                                              $\text{Sine} [0, 3.14]$  
$\psi$                                             $\text{Uniform} [0, 3.14]$  
$\phi_{\mathrm{JL}}$                               $\text{Uniform} [0, 6.28]$  
$\phi_{12}$                                        $\text{Uniform} [0, 6.28]$  
$\theta_{\mathrm{JN}}$                                $\text{Sine} [0, 3.14]$  
$m_f\ [M_{\odot}]$                                                        $-$  
$\chi_f$                                                                  $-$  
$v_{\rm kick}\ [{\rm km/s}]$                                              $-$



| Parameter                    | Posterior 90% CI                  | Prior                                           |
|:-----------------------------|:----------------------------------|:------------------------------------------------|
| $m_1\ [M_{\odot}]$           | ${84.54}_{-10.32}^{+11.37}$       | $-$                                             |
| $m_2\ [M_{\odot}]$           | ${52.20}_{-12.34}^{+11.93}$       | $-$                                             |
| $\mathcal{M}\ [M_{\odot}]$   | ${57.09}_{-6.14}^{+5.30}$         | $\text{UniformInComponentsChirpMass} [12, 400]$ |
| $q$                          | ${0.62}_{-0.19}^{+0.23}$          | $\text{UniformInComponentsMassRatio} [0.17, 1]$ |
| $a_1$                        | ${0.65}_{-0.30}^{+0.23}$          | $\text{Uniform} [0, 0.99]$                      |
| $a_2$                        | ${0.56}_{-0.36}^{+0.30}$          | $\text{Uniform} [0, 0.99]$                      |
| $\theta_1$                   | ${0.83}_{-0.40}^{+0.45}$          | $\text{Sine} [0, 3.14]$                         |
| $\theta_2$                   | ${1.13}_{-0.56}^{+0.73}$          | $\text{Sine} [0, 3.14]$                         |
| $\chi_{\mathrm{eff}}$        | ${0.32}_{-0.15}^{+0.14}$          | $-$                                             |
| $\chi_{\mathrm{p}}$          | ${0.47}_{-0.21}^{+0.24}$          | $-$                                             |
| $\alpha$                     | ${4.27}_{-1.26}^{+0.22}$          | $-$                                             |
| $\delta$                     | ${0.44}_{-0.68}^{+0.49}$          | $-$                                             |
| $t_c\ [s]$                   | ${1245035079.32}_{-0.02}^{+0.00}$ | $\text{Uniform} [1245035079.21, 1245035079.41]$ |
| $d_L\ [Mpc]$                 | ${2966.44}_{-956.83}^{+1071.36}$  | $\text{UniformSourceFrame} [100.0, 10000.0]$    |
| $\phi$                       | ${2.77}_{-1.85}^{+2.29}$          | $\text{Uniform} [0, 6.28]$                      |
| $\phi$                       | ${5.01}_{-1.89}^{+0.69}$          | $\text{Uniform} [0, 6.28]$                      |
| $\theta$                     | ${1.28}_{-0.15}^{+0.18}$          | $\text{Sine} [0, 3.14]$                         |
| $\psi$                       | ${1.43}_{-0.79}^{+0.96}$          | $\text{Uniform} [0, 3.14]$                      |
| $\phi_{\mathrm{JL}}$         | ${3.04}_{-2.37}^{+2.34}$          | $\text{Uniform} [0, 6.28]$                      |
| $\phi_{12}$                  | ${3.18}_{-2.26}^{+2.18}$          | $\text{Uniform} [0, 6.28]$                      |
| $\theta_{\mathrm{JN}}$       | ${2.16}_{-1.54}^{+0.55}$          | $\text{Sine} [0, 3.14]$                         |
| $m_f\ [M_{\odot}]$           | ${129.48}_{-9.28}^{+9.10}$        | $-$                                             |
| $\chi_f$                     | ${0.79}_{-0.07}^{+0.05}$          | $-$                                             |
| $v_{\rm kick}\ [{\rm km/s}]$ | ${687.60}_{-458.07}^{+804.75}$    | $-$                                             |


Lets make some plots!

In [5]:
# NRSurrogate corner plots

import os

param_sets = dict(
    mass=["mass_1", "mass_2", "chirp_mass", "mass_ratio"],
    spin=["a_1", "a_2", "tilt_1", "tilt_2"],
    effective_spin=["mass_ratio", "chi_eff", "chi_p"],
    sky_localisation=["luminosity_distance", "ra", "dec"],
    remnant=["final_mass", "final_spin", "final_kick"],
)
for name, params in param_sets.items():
    fname = f"GW190620_030421_{name}_corner.png"
    if not os.path.isfile(fname):
        fig = nrsur_result.plot_corner(params)
        fig.savefig(fname)

    if name == "remnant":
        continue

    # LVK-Comparison plots
    fname = f"GW190620_030421_compare_{name}_corner.png"
    if not os.path.isfile(fname):
        fig = nrsur_result.plot_lvk_comparison_corner(params)
        fig.savefig(fname)


## Corner Plots

### Mass


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190620_030421_mass_corner.png"](GW190620_030421_mass_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190620_030421_compare_mass_corner.png"](GW190620_030421_compare_mass_corner.png)
:::

::::



### Spin


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190620_030421_spin_corner.png"](GW190620_030421_spin_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190620_030421_compare_spin_corner.png"](GW190620_030421_compare_spin_corner.png)
:::

::::


### Effective Spin


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190620_030421_effective_spin_corner.png"](GW190620_030421_effective_spin_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190620_030421_compare_effective_spin_corner.png"](GW190620_030421_compare_effective_spin_corner.png)
:::

::::



### Sky-localisation


::::{tab-set}

:::{tab-item} NRSurrogate
:sync: key1

!["GW190620_030421_sky_localisation_corner.png"](GW190620_030421_sky_localisation_corner.png)
:::

:::{tab-item} LVK-Comparison
:sync: key2

!["GW190620_030421_compare_sky_localisation_corner.png"](GW190620_030421_compare_sky_localisation_corner.png)
:::

::::



### Remnant

!["GW190620_030421_remnant_corner.png"](GW190620_030421_remnant_corner.png)


## Waveform posterior-predictive plot

In [6]:
fname = f"GW190620_030421_waveform.png"
if not os.path.isfile(fname):
    fig = nrsur_result.plot_signal(outdir=".")


![waveform](GW190620_030421_waveform.png)

## Animations  
Here are some animations of the posterior.

|Spin       | 
|-----------|
| <iframe width="630" height="315" src="https://www.youtube.com/embed/zpvDh_-wbZI?autoplay=1&mute=1&playlist=zpvDh_-wbZI&loop=1" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>    |   

|Remnant       | 
|-----------|
| <iframe width="630" height="315" src="https://www.youtube.com/embed/9WIAMI6s-Ck?autoplay=1&mute=1&playlist=9WIAMI6s-Ck&loop=1" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>    | 


## Analysis configs
Below are the configs used for the analysis of this job.

In [7]:
nrsur_result.print_configs()

Tmax:                               10000
accounting:                         None
accounting_user:                    None
adapt:                              False
allow_tape:                         True
analysis_executable:                None
analysis_executable_parser:         None
autocorr_c:                         5.0
autocorr_tol:                       50.0
bilby_zero_likelihood_mode:         False
burn_in_nact:                       50.0
calibration_model:                  CubicSpline
calibration_prior_boundary:         reflective
catch_waveform_errors:              True
channel_dict:                       {L1:DCS-CALIB_STRAIN_CLEAN_SUB60HZ_C01, V1:Hrec_hoft_16384Hz}
check_point_deltaT:                 3600
clean:                              False
coherence_test:                     False
conda_env:                          None
condor_job_priority:                0
conversion_function:                None
create_summary:                     True
data_dict:                 

If you used this data, please [cite this work](../citation.md).

## Comments
Leave a comment below or in this [issue](https://github.com/avivajpeyi/nrsur_catlog/issues/new?title=GW190620_030421).
```{raw} html
<script src="https://utteranc.es/client.js"
        repo="avivajpeyi/nrsur_catalog"
        issue-term="GW190620_030421"
        theme="github-light"
        crossorigin="anonymous"
        async>
</script>
```